hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -257,13 +257,16 @@ private Table newHmsTable() {
     return newTable;
   }
 
-  private void setParameters(String newMetadataLocation, Table tbl, boolean hiveEngineEnabled) {
+  private void setParameters(String newMetadataLocation, Table tbl, TableMetadata meta, boolean hiveEngineEnabled) {
```

Comment:
Why not use the meta.properties instead of the full `TableMetadata`?

Reply:
Changed it